File: client-scalaz/src/main/scala/org/openapitools/client/api/WeatherForecastResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import WeatherForecastResponse._

case class WeatherForecastResponse (
  id: Option[Long],
forecastDate: Option[Long],
precipMM: Option[Double],
tempMaxC: Option[Integer],
tempMaxF: Option[Integer],
tempMinC: Option[Integer],
tempMinF: Option[Integer],
weatherCode: Option[Integer],
weatherDesc: Option[String],
weatherIconUrl: Option[String],
winddir16Point: Option[String],
winddirDegree: Option[Integer],
winddirection: Option[String],
windspeedKmph: Option[Integer],
windspeedMiles: Option[Integer])

object WeatherForecastResponse {
  import DateTimeCodecs._

  implicit val WeatherForecastResponseCodecJson: CodecJson[WeatherForecastResponse] = CodecJson.derive[WeatherForecastResponse]
  implicit val WeatherForecastResponseDecoder: EntityDecoder[WeatherForecastResponse] = jsonOf[WeatherForecastResponse]
  implicit val WeatherForecastResponseEncoder: EntityEncoder[WeatherForecastResponse] = jsonEncoderOf[WeatherForecastResponse]
}

File: client-clojure/src/sirqul_io_t_platform/specs/weather_forecast_response.clj
(ns sirqul-io-t-platform.specs.weather-forecast-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def weather-forecast-response-data
  {
   (ds/opt :id) int?
   (ds/opt :forecastDate) int?
   (ds/opt :precipMM) float?
   (ds/opt :tempMaxC) int?
   (ds/opt :tempMaxF) int?
   (ds/opt :tempMinC) int?
   (ds/opt :tempMinF) int?
   (ds/opt :weatherCode) int?
   (ds/opt :weatherDesc) string?
   (ds/opt :weatherIconUrl) string?
   (ds/opt :winddir16Point) string?
   (ds/opt :winddirDegree) int?
   (ds/opt :winddirection) string?
   (ds/opt :windspeedKmph) int?
   (ds/opt :windspeedMiles) int?
   })

(def weather-forecast-response-spec
  (ds/spec
    {:name ::weather-forecast-response
     :spec weather-forecast-response-data}))
